quantum/audio/audio_chibios_pwm.c

```diff
@@ -0,0 +1,195 @@
+/* Copyright 2020 Jack Humbert
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+Audio Driver: PWM
+
+the duty-cycle is always kept at 50%, and the pwm-period is adjusted to match the frequency of a note to be played back.
+
+this driver uses the chibios-PWM system to produce a squarewave on a given output pin.
+Either in software through a pwm callback and set/clear; or through the pwm hardware which directly toggles the pin via its alternate function. see your MCUs datasheet for pin can be driven by a timers T/IMx_CHy.
+
+ */
+
+
+/* STM32F103C Setup:
+halconf.h:
+#define HAL_USE_PWM                 TRUE
+#define HAL_USE_PAL                 TRUE
+#define HAL_USE_GPT                 TRUE
+
+mcuconf.h:
+#define STM32_PWM_USE_TIM1                  TRUE
+#define STM32_GPT_USE_TIM3                  TRUE
+
+used pin: PA8 (alternate0: Tim1_Ch1)
+
+from the datasheet for STM32F104C8B: alternate function of pin
```

Comment:
yes - the one on the ubiquitous bluepill boards
```suggestion
from the datasheet for STM32F103C8: alternate function of pin
```